src/library/scala/util/Either.scala

```diff
@@ -457,6 +457,35 @@ object Either {
   def cond[A, B](test: Boolean, right: => B, left: => A): Either[A, B] =
     if (test) Right(right) else Left(left)
 
+  /**
+   * Creates `FromLeftPartiallyApplied[B]` that given an argument of type `A`
+   * produces a `Left` typed as `Either[A,B]`
+   * {{
+   *   val l = Either.left[Int]("boo")
+   *   //l: Either[String, Int] = Left("boo")
+   * }}
+   */
+  def left[B]: FromLeftPartiallyApplied[B] = new FromLeftPartiallyApplied[B]
```

Comment:
It's my first time seeing this pattern, so it isn't obvious to me (and might not be obvious to other reviewers) what the downsides of this pattern are. I mean, obviously it's a downside that this extra type is involved (and shows up in the Scaladoc). Are there any concealed footguns here?

Reply:
that was my first time as well, but it looks safe enough, except the fact you mentioned - it appears in public
like you can do `Either.left[Int]` and get instance of `FromLeftPartiallyApplied[Int]`

Reply:
I'm fairly sure there's no footguns here :)